app/routes/companyInterest/components/CompanyInterestList.tsx

```diff
@@ -226,9 +233,25 @@ const CompanyInterestList = () => {
           <Flex column>
             <SelectInput
               name="form-semester-selector"
-              value={selectedSemesterFilterOption}
-              onChange={(clickedOption: SemesterOptionType) =>
-                setQueryValue('semesters')(String(clickedOption.id ?? ''))
+              value={{
+                label: currentCompanySemester
+                  ? semesterToText({
+                      semester: currentCompanySemester.semester,
+                      year: currentCompanySemester.year,
+                      language: 'norwegian',
+                    })
+                  : 'Vis alle semestre',
+                value: Number(currentCompanySemester?.id) ?? 9999,
+                year: currentCompanySemester?.year ?? 9999,
```

Comment:
💀 why not just the current year?

Reply:
If `currentCompanySemester` is `undefined` (which it really shouldn't since the default option is the current semester), the input should default to `"Vis alle semestre"`. Looking at the old implementation, the value defaulted to `''`, but `year` is now a `number`. Would `Number(currentCompanySemester?.year)` be better?